modules/tinymce/src/core/main/ts/textpatterns/core/BlockPattern.ts

```diff
@@ -55,10 +55,18 @@ const applyPattern = (editor: Editor, match: BlockPatternMatch): boolean => {
   return true;
 };
 
+const sortPatterns = <P extends Pattern>(patterns: P[]): P[] => Arr.sort(patterns, (a, b) => {
+  if (a.start.length === b.start.length) {
+    return 0;
+  }
+  return a.start.length > b.start.length ? -1 : 1;
```

Comment:
It doesn't need to be `1` and `-1`, only >0 and <0, so this whole thing can become
```suggestion
return b.start.length - a.start.length
```